node/src/effect/announcements.rs

```diff
@@ -217,6 +214,8 @@ pub enum BlockExecutorAnnouncement {
         /// The results of executing the deploys in this block.
         execution_results: HashMap<DeployHash, (DeployHeader, ExecutionResult)>,
     },
+    /// A block was requested to be executed, but it had been executed before.
+    BlockAlreadyExecuted(Block),
```

Comment:
Why is this an announcement? This should be a request – someone made a request to execute a block.

Reply:
This is not a response to a request. This is to signal that there _was_ a request to execute a block, but we found that block in storage, so it had been already executed and we won't executed it again. We need to announce it so that `LinearChainSync` can learn that it can make progress in such a case.

Reply:
I know, but some component had made a request to execute a block – either linear chain sync or era supervisor – and they should be notified about this via response and not an announcement.

Reply:
What if a _different_ component has to be notified, too? In this case the component(s) to be notified are independent of which component made the request. I don't think this is a good place to use the request-response pattern because of that.

Reply:
But it's not. You're guessing that maybe in a future we will have to do that but you may be as well wrong. And even if there will come a time when we find ourselves having to do that, there may be more than one solution for this. Maybe one won't involve relying on announcements.